### ISSUE TITLE
make-check.py not in 5.3.11 distributable

### ISSUE THREAD

**Issue by errantlinguist** [opencog/link-grammar#449]
On the AbiWord website, the current "stable" version is listed as 5.3.11 (see [listed download link](http://www.abisource.com/downloads/link-grammar/5.3.11/link-grammar-5.3.11.tar.gz)). In this version, the usage of the script file `make-check.py` is mentioned in the `README` for using Python bindings, but this file is not included in the distributable.

**Comment by ampli** [comment]
Thanks.
I have just set a fix.
Meanwhile you can use [make-check.py](https://github.com/opencog/link-grammar/blob/83e5c9bfe07a3dd816d4df2fa2b7579658c85ef9/msvc14/make-check.py) from the GitHub repository (save the "Raw" version).

**Comment by errantlinguist** [comment]
Should the script work without any modification?-- I can't test the setup on a Linux machine at the moment, but after building the parser using Visual Studio/C++, I get

    LINK_GRAMMAR_DATA=..
    Traceback (most recent call last):
      File "C:\Users\tcshore\src\link-grammar-5.3.11\msvc14\..\bindings\python-examples\tests.py", line 20, in <module>
        from linkgrammar import (Sentence, Linkage, ParseOptions, Link, Dictionary,
      File "C:\Users\tcshore\src\link-grammar-5.3.11\bindings\python\linkgrammar.py", line 13, in <module>
        import clinkgrammar as clg
      File "C:\Users\tcshore\src\link-grammar-5.3.11\bindings\python\clinkgrammar.py", line 28, in <module>
    _clinkgrammar = swig_import_helper()
      File "C:\Users\tcshore\src\link-grammar-5.3.11\bindings\python\clinkgrammar.py", line 20, in swig_import_helper
        import _clinkgrammar
    ImportError: DLL load failed: %1 is not a valid Win32 application.

Using the batch-file demo file `link-parser.bat` works without any problems, however, so must something not be wrong with SWIG and/or the Python bindings themselves?

**Comment by ampli** [comment]
Yes, it should work without any modifications (I have just checked it on 5.3.11, with both Python2 & 3).
My tests environment:
- Visual Studio 2015.
- x64 machine.
- Link grammar compiled as x64 (I tried both Release and Debug).
- Python versions:
 - `Python 2.7.12 (v2.7.12:d33e0cf91556, Jun 27 2016, 15:24:40) [MSC v.1500 64 bit (AMD64)] on win32`
 - `Python 3.4.4 (v3.4.4:737efcadf5a6, Dec 20 2015, 20:20:57) [MSC v.1600 64 bit (AMD64)] on win32`
- Command: `make-check.py x64\Debug\Python2` (tried all cobinations of Debug/Release & Python2/Python3)

Is your test environment the same?

**Comment by ampli** [comment]
Can you please also include the output of `make-check.py` before `LINK_GRAMMAR_DATA=..`?
Maybe it will help to find the problem.

**Comment by errantlinguist** [comment]
Thanks for the extra info: I found out that I was mistakenly trying to run the bindings using a 64-bit Python implementation despite that I build the bindings for Win32 -- this solved the error above but now I have a more "conventional"-looking message:

    PYTHONPATH=C:\Users\tcshore\src\link-grammar-5.3.11\msvc14\..\bindings\python;C:\Users\tcshore\src\link-grammar-5.3.11\msvc14\Win32\Debug\Python2
    Issuing command: C:\Python27\python.exe  C:\Users\tcshore\src\link-grammar-5.3.11\msvc14\..\bindings\python-examples\tests.py
    Running by: C:\Python27\python.exe
    Running C:\Users\tcshore\src\link-grammar-5.3.11\msvc14\..\bindings\python-examples\tests.py in: C:\Users\tcshore\src\link-grammar-5.3.11\msvc14
    PYTHONPATH=C:\Users\tcshore\src\link-grammar-5.3.11\msvc14\..\bindings\python;C:\Users\tcshore\src\link-grammar-5.3.11\msvc14\Win32\Debug\Python2
    srcdir=None
    LINK_GRAMMAR_DATA=..
    Traceback (most recent call last):
      File "C:\Users\tcshore\src\link-grammar-5.3.11\msvc14\..\bindings\python-examples\tests.py", line 20, in <module>
        from linkgrammar import (Sentence, Linkage, ParseOptions, Link, Dictionary,
      File "C:\Users\tcshore\src\link-grammar-5.3.11\bindings\python\linkgrammar.py", line 13, in <module>
        import clinkgrammar as clg
      File "C:\Users\tcshore\src\link-grammar-5.3.11\bindings\python\clinkgrammar.py", line 28, in <module>
    _clinkgrammar = swig_import_helper()
      File "C:\Users\tcshore\src\link-grammar-5.3.11\bindings\python\clinkgrammar.py", line 20, in swig_import_helper
        import _clinkgrammar
    ImportError: DLL load failed: The specified module could not be found.

This is using 

- Visual Studio 2015
- x64 machine
- Link Grammar compiled for Win32 (Debug target) because the regex library used is 32-bit ([GnuWin32](http://gnuwin32.sourceforge.net/packages/regex.htm#download))
- Python 2.7.12 (v2.7.12:d33e0cf91556, Jun 27 2016, 15:19:22) [MSC v.1500 32 bit (Intel)] on win32
- Command: `make-check.py Win32\Debug\Python2`

**Comment by ampli** [comment]
It looks as if one of the DLLs `_clinkgrammar.pyd` or `link-grammar-x86.dll` is not found.
`make-check-py` puts the directory of `_clinkgrammar.pyd` in`PYTHONPATH` and the directory of `link-grammar-x86.dll` in PATH (but it prints only `PYTHONPATH`).

Please try the following command at the `cmd` prompt, in the top-level link-grammar source directory:

`dir/s _clinkgrammar.pyd`
It should be found in `msvc14\Win32\Debug\Python2`.

`dir/s link-grammar-x86.dll`
It should be found in `msvc14\Win32\Debug`.

(I suppose you make your checks on the same computer in which you make the compilations.)

**Comment by errantlinguist** [comment]
Sadly, I could indeed find both `clinkgrammar.pyd` and `link-grammar-x86.dll` in the directories you mentioned, but I'm still getting the "DLL not found" error.

**Comment by ampli** [comment]
In which PATH component `regex2.dll` is found?
(I supposed it is in your PATH because I understood `link-parser` works fine.)

**Comment by errantlinguist** [comment]
Actually, at that moment it's not in the path; In order to get `link-parser.bat` to work, I edited it from the original to have the following:

    REM DLLs central directory
    set "PATH=%PATH%;%GNUREGEX_DIR%\bin"

Sorry, was I supposed to add the regex library directory to the global path for Windows (and not just for the path used for running `link-parser`?-- I guess I may have misunderstood the readme instructions:

> The last step of the "LinkGrammarExe" project creates "link-parser.bat".  Copy
it to somewhere in your PATH and then customize it if needed (don't customize
it in-place, since rebuilding would overwrite it). Especially, add to PATH the
directory in which the regex DLL resides (see the line containing "DLLs"). If
you would like to display the wordgraph (see below) set also the PATH for
"dot.exe" if it is not already in your PATH.

**Comment by ampli** [comment]
You understood it correctly.
Indeed,  `make-check.py`, which was indroduced much after `link-parser.bat`, just supposes the regex DLL is in PATH. I neglected to consider this point when I wrote it, this is a bug...

I think I will change `make-check.py` to automatically include the regex DLL in PATH (e.g by using a User Macro GNUREGEX_DLL). In that case `link-parser.bat` can automatically include it too, without a need for a manual setup.

So for now there is a need to add the regex DLL directory in the global PATH or in `make-check.py`  (in `os.environ["PATH"]` ).
